lib/spack/spack/rewiring.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+import re
+import shutil
+import tempfile
+from collections import OrderedDict
+
+import spack.binary_distribution as bindist
+import spack.hooks
+import spack.paths
+import spack.relocate as relocate
+import spack.stage
+import spack.store
+
+
+def _relocate_spliced_links(links, orig_prefix, new_prefix):
+    """Re-linking function which differs from `relocate.relocate_links` by
+    reading the old link rather than the new link, since the latter wasn't moved
+    in our case. This still needs to be called after the copy to destination
+    because it expects the new directory structure to be in place."""
+    for link in links:
+        link_target = os.readlink(os.path.join(orig_prefix, link))
+        link_target = re.sub('^' + orig_prefix, new_prefix, link_target)
+        new_link_path = os.path.join(new_prefix, link)
+        os.unlink(new_link_path)
+        os.symlink(link_target, new_link_path)
+
+
+def rewire(spliced_spec):
+    """Given a spliced spec, this function conducts all the rewiring on all
+    nodes in the DAG of that spec."""
+    assert spliced_spec.spliced
+    for spec in spliced_spec.traverse(order='post', root=True):
+        if not spec.build_spec.package.installed:
+            # TODO: May want to change this at least for the root spec...
+            spec.build_spec.package.do_install(force=True)
+        if spec.build_spec is not spec and not spec.package.installed:
+            explicit = spec is spliced_spec
+            rewire_node(spec, explicit)
+
+
+def rewire_node(spec, explicit):
+    """This function rewires a single node, worrying only about references to
+    its subgraph. Binaries, text, and links are all changed in accordance with
+    the splice. The resulting package is then 'installed.'"""
+    tempdir = tempfile.mkdtemp()
+    # copy anything installed to a temporary directory
+    shutil.copytree(spec.build_spec.prefix,
+                    os.path.join(tempdir, spec.dag_hash()))
+
+    spack.hooks.pre_install(spec)
+    # compute prefix-to-prefix for every node from the build spec to the spliced
+    # spec
+    prefix_to_prefix = OrderedDict({spec.build_spec.prefix: spec.prefix})
+    for build_dep in spec.build_spec.traverse(root=False):
+        prefix_to_prefix[build_dep.prefix] = spec[build_dep.name].prefix
+
+    manifest = bindist.get_buildfile_manifest(spec.build_spec)
+    platform = spack.platforms.by_name(spec.platform)
+    if manifest.get('text_to_relocate'):
```

Comment:
```suggestion
    text_to_relocate = maniifest.get('text_to_relocate', None):
    if text_to_relocate:
```
Then you can use it inside the block wiithout needing a separate lookup

Reply:
This has been changed accordingly in both the `text` and `binary` cases.